src/txmempool.h

```diff
@@ -138,6 +179,15 @@ class CTxMemPool
     void ApplyDeltas(const uint256 hash, double &dPriorityDelta, CAmount &nFeeDelta);
     void ClearPrioritisation(const uint256 hash);
 
+    /** Build a list of transaction (hashes) to remove such that:
+     *  - The list is consistent (if a parent is included, all its dependencies are included as well).
+     *  - Removing said list will reduce the DynamicMemoryUsage below sizelimit.
+     *  - At most maxfeeremove worth of fees will be removed.
+     *  - No transaction whose hash is in prot will be removed.
+     */
+    bool StageTrimToSize(size_t sizelimit, const CAmount& maxfeeremove, const CFeeRate& repfeerate, const std::set<uint256>& prot, std::set<uint256>& stage, CAmount& totalfeeremoved, size_t& totalsizeremoved);
+    void RemoveStaged(std::set<uint256>& stage);
```

Comment:
The rationale here is that I would want the staged set to become a separate class, which tracks statistics of what is being removed etc. That would make it also safe to not expose removeUnchecked (way faster than remove, because it already contains all dependencies), but a RemoveStaged method would need to remain.